src/Centreon/Domain/PlatformTopology/PlatformTopology.php

```diff
@@ -0,0 +1,234 @@
+     * @var string Server parent IP
+     */
+    private $serverParentAddress;
+    /**
+     * @var int Server parent id
+     */
+    private $serverParentId;
+
+    /**
+     * @return int
+     */
+    public function getServerType(): int
+    {
+        return $this->serverType;
+    }
+
+    /**
+     * @param int $serverType server type
+     *      SERVER_TYPE_CENTRAL = 0
+     *      SERVER_TYPE_POLLER  = 1
+     *      SERVER_TYPE_REMOTE  = 2
+     *      SERVER_TYPE_MAP     = 3
+     *      SERVER_TYPE_MBI     = 4
+     *
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerType(int $serverType): self
+    {
+        // The API should not be used to add a Central to another Central
+        if (self::SERVER_TYPE_CENTRAL === $serverType) {
+            throw new PlatformTopologyException(
+                sprintf(
+                    _("You cannot link the Central '%s'@'%s' to another Central"),
+                    $this->getServerName(),
+                    $this->getServerAddress()
+                ),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        // Check if the server_type is available
+        $availableServerType = [
+            self::SERVER_TYPE_POLLER,
+            self::SERVER_TYPE_REMOTE,
+            self::SERVER_TYPE_MAP,
+            self::SERVER_TYPE_MBI
+        ];
+        if (!in_array($serverType, $availableServerType)) {
+            throw new PlatformTopologyException(
+                sprintf(
+                    _("The type of platform '%s'@'%s' is not consistent"),
+                    $this->getServerName(),
+                    $this->getServerAddress()
+                ),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $this->serverType = $serverType;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getServerName(): string
+    {
+        return $this->serverName;
+    }
+
+    /**
+     * @param string $serverName
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerName(string $serverName): self
+    {
+        $serverName = filter_var($serverName, FILTER_SANITIZE_STRING);
+        if (empty($serverName)) {
+            throw new PlatformTopologyException(
+                _('The name of the platform is not consistent'),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $this->serverName = $serverName;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getServerAddress(): string
+    {
+        return $this->serverAddress;
+    }
+
+    /**
+     * @param string $serverAddress
+     *
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerAddress(string $serverAddress): self
+    {
+        $this->serverAddress = $this->checkIpAddress(self::KIND_SERVER, $serverAddress);
```

Comment:
Be careful, your method does not only test IP addresses. Please change its name.

Reply:
it does check only IP address, as the DNS must be resolved before being validated